web/src/message_util.ts

```diff
@@ -112,3 +116,41 @@ export function get_topics_for_message_ids(message_ids: number[]): Map<string, [
     }
     return topics;
 }
+
+export function can_previous_pms_exist(recipient_ids_string: string): {
+    allow_local_echo: boolean;
+    can_previous_pms_exist: boolean;
+} {
+    // Check if there are any previous messages in the DM conversation.
+    const previous_direct_messages_exist = recent.has_conversation(recipient_ids_string);
```

Comment:
This variable name is confusing. The check here is whether __we know__ some exist.

Maybe `const have_conversation_in_cache = ...`